### PR TITLE
Add clarification to softmax and log_softmax regarding masked-out elements.

### PR DESCRIPTION
**Pull request by carlosgmartin**
Adds a minor clarification to the documentation for [`softmax`](https://docs.jax.dev/en/latest/_autosummary/jax.nn.softmax.html) and [`log_softmax`](https://docs.jax.dev/en/latest/_autosummary/jax.nn.log_softmax.html) explaining the output behavior for masked-out elements.

### CODE REVIEW

**Review by jakevdp — APPROVED** [review]
Thanks!
